Tests.Unit.Core.ApplicationServices/ApplicationServices/SystemUsage/ItSystemUsageWriteServiceTest.cs

```diff
@@ -1087,6 +1035,304 @@ public void Can_Update_Roles_To_Remove_Them()
             AssertRemoveRoleCalledOnce(itSystemUsage, rightToRemove1.Role.Uuid, rightToRemove1.User.Uuid);
             AssertAssignRoleCalledOnce(itSystemUsage, newRight.Role.Uuid, newRight.User.Uuid);
         }
+
+        [Fact]
+        public void Can_Create_With_GDPR()
+        {
+            //Arrange
+            var (systemUuid, organizationUuid, transactionMock, organization, itSystem, itSystemUsage) = CreateBasicTestVariables();
+
+            SetupBasicCreateThenUpdatePrerequisites(organizationUuid, organization, systemUuid, itSystem, itSystemUsage);
+            var sensitiveDataTypeUuids = Many<Guid>().Distinct().ToList();
+            var registerTypeUuids = Many<Guid>().Distinct().ToList();
+
+            ExpectUpdateSensitiveDataTypesReturns(itSystemUsage, sensitiveDataTypeUuids, sensitiveDataTypeUuids.Select(uuid => new SensitivePersonalDataType { Uuid = uuid, Id = A<int>() }).ToList());
+            ExpectUpdateRegisterTypesReturns(itSystemUsage, registerTypeUuids, registerTypeUuids.Select(uuid => new RegisterType { Uuid = uuid, Id = A<int>() }).ToList());
+
+            var purpose = A<string>();
+            var businessCritical = A<DataOptions?>();
+            var hostedAt = A<HostedAt?>();
+            var directoryDoc = A<NamedLink>();
+            var sensitiveDataLevels = Many<SensitiveDataLevel>().Distinct().ToList();
+            var technicalPrecautionsInPlace = A<DataOptions?>();
+            var technicalPrecautions = Many<TechnicalPrecaution>().Distinct().ToList();
+            var technicalPrecautionsDocumentation = A<NamedLink>();
+            var userSupervision = A<DataOptions?>();
+            var supervisionDate = A<DateTime?>();
+            var supervisionDoc = A<NamedLink>();
+            var riskAssessmentConducted = A<DataOptions?>();
+            var riskAssessmentDate = A<DateTime?>();
+            var riskAssessmentDoc = A<NamedLink>();
+            var riskAssessmentNotes = A<string>();
+            var riskAssessmentResult = A<RiskLevel?>();
+            var dpiaConducted = A<DataOptions?>();
+            var dpiaDate = A<DateTime?>();
+            var dpiaDoc = A<NamedLink>();
+            var retentionPeriod = A<DataOptions?>();
+            var nextEvaluationDate = A<DateTime?>();
+            var evaluationFrequency = A<int?>();
+            var gdprInput = new UpdatedSystemUsageGDPRProperties
```

Comment:
Det var godt nok en lang opsætning og senere Assert. Ved godt de kun bruges en gang, men for at have lettere ved at læse testen kunne de godt trækkes ud i metoder.

Reply:
jeg bruger værdierne i assertions længere nede for ikke at skulle lave input.Purpose.Value.Value.Value :-)